app/models/concerns/orchestration/ssh_provision.rb

```diff
@@ -82,6 +82,10 @@ def setSSHProvision
       Host.find(id).built
       respond_to?(:initialize_puppetca,true) && initialize_puppetca && delAutosign if puppetca?
     else
+      if Setting[:clean_up_failed_deployment]
+        logger.info "Deleting host #{name} because of non zero exit code of deployment script."
+        Host.find(id).destroy
+      end
       raise ::Foreman::Exception.new(N_("Provision script had a non zero exit, removing instance"))
```

Comment:
The "removing instance" comment here is peculiar, I wonder if it changed in the past.  Could you remove that part?  We should to rely on the log message you added to say whether it did, or didn't remove the VM.